src/libraries/System.Runtime/tests/System/AppContext/AppContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Tests
+{
+    public partial class AppContextTests
+    {
+        [Theory]
+        [InlineData("AppContext_Case1", 123)]
+        [InlineData("AppContext_Case2", "")]
+        [InlineData("AppContext_Case3", null)]
+        public void AppContext_GetSetDataTest(string dataKey, object value)
+        {
+            // Set data
+            AppContext.SetData(dataKey, value);
+
+            // Get previously set data
+            object actual = AppContext.GetData(dataKey);
+
+            // Validate data equality
+            Assert.Equal(value, actual);
+
```

Comment:
```suggestion
```
The instance equality implies data equality. No need to validate the data equality.

Reply:
Oh right. Thanks :)